### ISSUE TITLE
Bug decompressing small chunks

### ISSUE THREAD

**Issue by aleixalcacer** [Blosc/c-blosc2#300]
**Describe the bug**
There is a bug when a really small chunk is decompressed using multiple threads.

**To Reproduce**
```
#include <blosc2.h>

int main() {
    blosc_init();

    int16_t nthreads = 2;
    int64_t data_size = 4;
    int32_t blocksize = 1;
    int32_t itemsize = 1;

    // Init data
    uint8_t *data = malloc(data_size * itemsize);
    for (int i = 0; i < data_size; ++i) {
        data[i] = i;
    }

    // Create compression context
    blosc2_cparams cparams = BLOSC2_CPARAMS_DEFAULTS;
    cparams.nthreads = nthreads;
    cparams.blocksize = blocksize;
    cparams.typesize = itemsize;

    blosc2_context *cctx = blosc2_create_cctx(cparams);

    // Create decompression context
    blosc2_dparams dparams = BLOSC2_DPARAMS_DEFAULTS;
    dparams.nthreads = nthreads;
    blosc2_context *dctx = blosc2_create_dctx(dparams);

    // Compress data into chunk
    int64_t chunk_size = data_size + BLOSC_MAX_OVERHEAD;
    uint8_t *chunk = malloc(chunk_size);
    int cb = blosc2_compress_ctx(cctx, data, data_size, chunk, chunk_size);
    printf("cbytes: %d\n", cb);

    // Decompress chunk into data2
    uint8_t *data2 = malloc(data_size);
    int nb = blosc2_decompress_ctx(dctx, chunk, chunk_size, data2, data_size);
    printf("nbytes: %d\n", nb);

    // Print data and data2
    for (int i = 0; i < data_size; ++i) {
        printf("%6d", data[i]);
    }
    printf("\n");

    for (int i = 0; i < data_size; ++i) {
        printf("%6d", data2[i]);
    }
    printf("\n");

    // Free contexts
    blosc2_free_ctx(cctx);
    blosc2_free_ctx(dctx);

    blosc_destroy();

    return 0;
}
```

**Expected behavior**
`data` and `data2` should be equals.

**Comment by aleixalcacer** [comment]
Add this example to the test suite

**Comment by FrancescAlted** [comment]
Fixed in #329.
